src/components/SkillsOverview.tsx

```diff
@@ -1,14 +1,19 @@
-import React from "react"
-import { skills } from "../data/skills";
+import React from "react";
 import { Skill } from "../types/skill";
 import { Link } from "react-router-dom";
 
-function SkillsOverview(){
-const tablerows = skills.map((skill: Skill) => (
+function SkillsOverview() {
+  const updatedSkillList: string | null = window.localStorage.getItem("skills");
```

Comment:
Achtung es fängt der erste Teil eines Refactorings an. Die anderen Teile schreibe ich aber noch nicht, weil wir es Schritt für Schritt machen.

Sinn des erster Teils: 
a) Duplizierten Code eliminieren
b) Verteilen von Verantwortung (Welche Datei *.tsx, *.js, *.ts ist für was zuständig?)
c) Das verteilen von Verantwortung hilft dabei in sehr großen Projekten den Überblick zu bewahren. Denn man kann dann allein vom Namen der Datei immer sagen, was da drin ist. (Beispiel: skills.api.ts - "ah ok, in dieser Datei geht es NUR um die Kommunikation mit dem Backend und NUR für Skills")

Code:
```
winow.localStorage.getItem
...
JSON.parse
```

a) Dieser Code ist dupliziert (selbsterklärend).

b) Du musst hier in SkillsOverview ein JSON parsen. Weil die Skills aus dem localStorage kommen. Aber eigentlich interessiert es dich hier nicht. Hier in SkillsOverview.tsx, hast du als Entwickler das Ziel eine hübsche Benutzeroberfläche im Browser aufzubauen. Eine Übersicht von Skills. Dafür brauchst du für diese Übersicht die aktuellen Skills. Ok. Aber, dass die Skills aus dem local storage kommen, musst du hier gar nicht wissen. Anders gesagt es interessiert dich nicht. Nochmal anders gesagt die SkillsOverview.tsx will gar nicht wissen woher die Skills genau kommen.

Besser:
```
const skills = skillService.getSkills();
skills.map ....
```

Jetzt mit diesem Code holt die SkillsOverview.tsx skills. Aber woher die kommen steht hier nicht mehr. Der skillService kümmert sich um den localStorage und auch um das parsen. Spoiler: wir schmeißen später den localStorage weg und setzen eine Datenbank ein. Cool - dann müssen wir SkillsOverview ** hinterher nicht mehr anpassen**.


c) 
Also dann: Bitte eine neue Datei "skill.service.ts" anlegen.

`
/**
 * Diese Funktion gibt eine Liste der aktuellen Skills zurück. Wenn keine Skills vorhanden sind, dann ist die Liste leer.
 */
export function getSkills(): Skill[] {

... logik wegen localStorage

  return .....( ein Array von Typ Skill[] ) ....
}
`

Wieder ein Vorteil: 
Cool, derjenige, der die Funktion getSkills() aufruft (also SkillsOverview.tsx) muss sich nicht ums parsen kümmern und auch keine null checks durchführen. Der Aufrufer bekommt entweder ein leeres Array oder ein Array mit Skills.Das macht weniger Kopfsschmerzen.